### DISCUSSION TITLE
Airflow Breeze on WSL2 - permission issue and development question

### DISCUSSION THREAD

**Discussion by hrvoje-intuita** [apache/airflow#29370] [Q&A]
Hi,

When running "code ." from WSL2 distro like Ubuntu I can get whole development in Python up and running without problems in Visual Studio Code in Windows.
My questions is - how to achieve same in Airflow?
Breeze works fine with :
`breeze start-airflow --python 3.10  --load-example-dags --backend postgres --postgres-version 14 --use-airflow-version 2.5.0 --airflow-constraints-reference https://raw.githubusercontent.com/apache/airflow/constraints-2.5.0/constraints-3.10.txt --airflow-extras pandas,amazon,postgres,sftp`

But then when I enter `code . ` in 
`root@d58a6f893c0b:/opt/airflow# code .
`
bash: code: command not found. It seems I cannot start it like that from inside docker.

`code .` works from airflow folder in WSL2 Ubuntu if I don't use breeze but that seems besides point. In that case VS Code detects Dockerfile and tries to run it but I fires up sqlite and other stuff I don't need. It seems like than I have to manually hassle through docker file to setup whole thing which is what Breeze suppose to be here for. 
And if  I don't use Docker file than I can start development in WSL2 in which case I get permission problems like this:
https://stackoverflow.com/questions/55274601/airflow-valueerror-unable-to-configure-handler-processor-wasb-logger

For which I don't have solution because none of those suggestions work.

**Comment by hussein-awala** [comment]
> But then when I enter code .  in
root@d58a6f893c0b:/opt/airflow# code . 
bash: code: command not found

This is normal, because Vs Code is not installed in Airflow docker image

> VS Code detects Dockerfile and tries to run it but I fires up sqlite and other stuff I don't need.
And if I don't use Docker file than I can start development in WSL2 in which case I get permission problems like this

If you want to contribute to Airflow, the best way to run Airflow unit tests and have the updates immediately is using Breeze, but you don't need to write the code inside it, where the code is shared with the container via docker volumes.

But if you want to run Airflow to develop dags/operators for your own project, the best approach is using [docker-compose](https://github.com/apache/airflow/blob/main/docs/apache-airflow/howto/docker-compose/index.rst) to build a custom image and run all the services which you need to test the dags. And here also you don't need to write the code inside the container, you can just update the [volumes](https://github.com/apache/airflow/blob/main/docs/apache-airflow/howto/docker-compose/docker-compose.yaml#L64-L67) based on your files structure, and develop form you host via Vs Code.

**Reply by hrvoje-intuita** [reply]
Yeah, I got that, I was just thinking that it would be nice if there is a way to spin up Breeze image of a choice and do own project development based on that particular docker image. After you develop your project you can than test it with Breeze which would be great. 
This would enable local and dev part of CI/CD to be run locally before deployment to dev. 

I was thinking of running docker composer image on WSL2 and mounting dags and plugins in same folder that Breeze uses so I can latter test same image with Breeze. Would that be ok?

**Reply by potiuk** [reply]
Sure. You can try. Airflow itself uses Breeze image for CI.
